File: eratostenovo_sito/eratostenovo_sito.py
#!/usr/env/python3

def eratostenovo_sito(n):
  assert n > 1, "broj mora bit veci od 1"
  prime = [True for i in range(n+1)]
  prime[0] = prime[1] = False
  p = 2
  while (p**2 <= n):
    if prime[p] == True:
      # eliminiraj sve visekratnike trenutnog broja
      for i in range(p*2, n+1, p):
        prime[i] = False
    p += 1
  
  ret = [n for n in range(len(prime)) if prime[n] == True]
  return ret

if __name__ == "__main__":
  print(eratostenovo_sito(int(input())))

File: eratostenovo_sito/eratostenovo_sito.cpp
#include <iostream>
#include <vector>
#include <cassert>
#include <cstring>
using namespace std;

void eratostenovo_sito(int n){
  assert (n > 1); // broj mora bit veci od 1
  bool prime[n+1];
  memset(prime, true, sizeof(prime));
  for (int p = 2; p*p<=n; ++p){
    if (prime[p] == true){
      // eliminiraj sve visekratnike trenutnog broja
      for (int i = p*p; i <= n; i+=p){
        prime[i] = false;
      }
    }
  }
  for (int p = 2; p <= n; ++p){
    if (prime[p]){
      cout << p << " ";
    }
  }
  cout << "\n";
}

int main(){
  int n;
  cin >> n;
  eratostenovo_sito(n);
  return 0;
}
